test/gtest/db_sync.cpp

```diff
@@ -0,0 +1,606 @@
+    else if(data_type == "INT8x4")
+        return miopenInt8x4;
+    else if(data_type == "INT32")
+        return miopenInt32;
+    else if(data_type == "BF16")
+        return miopenBFloat16;
+    else if(data_type == "FP64")
+        return miopenDouble;
+    throw std::runtime_error("Invalid data type in find db key");
+}
+void ParseProblemKey(const std::string& key_, conv::ProblemDescription& prob_desc)
+{
+    std::string key = key_;
+    const auto opt  = SplitDelim(key, '_');
+    int group_cnt   = 1;
+    conv::Direction dir;
+    miopenTensorLayout_t in_layout, wei_layout, out_layout;
+    size_t out_h, out_w, in_channels, out_channels, in_h, in_w, batchsize, fil_h, fil_w;
+    int pad_h, pad_w, conv_stride_h, conv_stride_w, dil_h, dil_w;
+    miopenDataType_t precision;
+    TensorDescriptor in{};
+    TensorDescriptor wei{};
+    TensorDescriptor out{};
+    ConvolutionDescriptor conv;
+    if(opt.size() >= 2)
+    {
+        key = opt[0];
+        ASSERT_TRUE(StartsWith(opt[1], "g"));
+        group_cnt = std::stoi(RemovePrefix(opt[1], "g"));
+    }
+    else
+        ASSERT_TRUE(opt.size() == 1); // either there is one optional args or there is none
+    // 2d or 3d ?
+    const auto is_3d = [&]() {
+        const auto pat_3d = std::regex{"[0-9]x[0-9]x[0-9]"};
+        return std::regex_search(key, pat_3d);
+    }();
+    const auto attrs = SplitDelim(key, '-');
+    const auto sz    = attrs.size();
+    dir              = GetDirectionFromString(attrs[sz - 1]);
+    precision        = GetDataTypeFromString(attrs[sz - 2]);
+    if(!is_3d)
+    {
+        ASSERT_TRUE(sz == 15 || sz == 17);
+        std::tie(in_layout, wei_layout, out_layout) = [&]() {
+            if(sz == 15) // same layout for all tensors
+                return std::tuple{GetLayoutFromString(attrs[12]),
+                                  GetLayoutFromString(attrs[12]),
+                                  GetLayoutFromString(attrs[12])};
+            else if(sz == 17)
+                return std::tuple{GetLayoutFromString(attrs[12]),
+                                  GetLayoutFromString(attrs[13]),
+                                  GetLayoutFromString(attrs[14])};
+            throw std::runtime_error{"FDB key parsing error"};
+        }();
+
+        // if(dir == conv::Direction::Forward)
+        {
+            in_channels  = std::stoi(attrs[0]);
+            in_h         = std::stoi(attrs[1]);
+            in_w         = std::stoi(attrs[2]);
+            out_channels = std::stoi(attrs[4]);
+            out_h        = std::stoi(attrs[5]);
+            out_w        = std::stoi(attrs[6]);
+        }
+#if 0
+        else
+        {
+            out_channels = std::stoi(attrs[0]);
+            out_h        = std::stoi(attrs[1]);
+            out_w        = std::stoi(attrs[2]);
+            in_h         = std::stoi(attrs[5]);
+            in_w         = std::stoi(attrs[6]);
+            in_channels  = std::stoi(attrs[4]);
+        }
+#endif
```

Comment:
To my understanding the input/output in the fdb key are switched for forward vs backwards directions. However, in this case I discovered doing so would construct invalid Problem Descriptions. @DrizztDoUrden can you help me understand what is going on here ?